jkube-kit/config/image/src/test/java/org/eclipse/jkube/kit/config/image/ImageNameTest.java

```diff
@@ -179,42 +154,76 @@ public void namesUsedByDockerTests() {
     }
 
     @Test
-    public void testImageNameWithUsernameHavingPeriods() {
+    void testImageNameWithUsernameHavingPeriods() {
         // Given
         String name = "roman.gordill/customer-service-cache:latest";
 
         // When
         ImageName imageName = new ImageName(name);
 
         // Then
-        assertNotNull(imageName);
-        assertEquals("roman.gordill", imageName.getUser());
-        assertEquals("roman.gordill/customer-service-cache", imageName.getRepository());
-        assertEquals("latest", imageName.getTag());
-        assertNull(imageName.getRegistry());
+        assertThat(imageName).isNotNull();
+        assertThat(imageName.getUser()).isEqualTo("roman.gordill");
+        assertThat(imageName.getRepository()).isEqualTo("roman.gordill/customer-service-cache");
+        assertThat(imageName.getTag()).isEqualTo("latest");
+        assertThat(imageName.getRegistry()).isNull();
     }
 
     @Test
-    public void testImageNameWithNameContainingRegistryAndName() {
+    void testImageNameWithNameContainingRegistryAndName() {
         // Given
         String name = "foo.com:5000/customer-service-cache:latest";
 
         // When
         ImageName imageName = new ImageName(name);
 
         // Then
-        assertNotNull(imageName);
-        assertNull(imageName.getUser());
-        assertEquals("foo.com:5000", imageName.getRegistry());
-        assertEquals("customer-service-cache", imageName.getRepository());
-        assertEquals("latest", imageName.getTag());
+        assertThat(imageName).isNotNull();
+        assertThat(imageName.getUser()).isNull();
+        assertThat(imageName.getRegistry()).isEqualTo("foo.com:5000");
+        assertThat(imageName.getRepository()).isEqualTo("customer-service-cache");
+        assertThat(imageName.getTag()).isEqualTo("latest");
+    }
+
+    // TODO: Meaningful display name
+    @ParameterizedTest
+    @MethodSource("registryNamingTestData")
+    void testRegistryNaming(String imageName, String registry, String expected) {
```

Comment:
@manusa please suggest a meaningful name for this test

Reply:
@rohanKanojia please suggest descriptive name for this test

Reply:
I think maybe we can try this
```suggestion
    @DisplayName("getFullName should add registry to image name when valid registry is provided")
    void getFullName_whenRegistryProvided_shouldReturnImageNameWithRegistry(String imageName, String registry, String expected) {
```